poly/src/evaluations/multivariate/multilinear/sparse.rs

```diff
@@ -77,12 +77,6 @@ impl<F: Field> SparseMultilinearExtension<F> {
             }
             map.entry(index).or_insert(F::rand(rng));
         }
-        let mut buf = Vec::new();
-        for (arg, v) in map.iter() {
-            if *v != F::zero() {
-                buf.push((*arg, *v));
-            }
-        }
```

Comment:
Why was this loop removed?

Reply:
because buf is never used after this so right now this loop is kind of useless

Reply:
I see...